### DISCUSSION TITLE
Happy 2022!  No, @robla hasn't made progress

### DISCUSSION THREAD

**Discussion by robla** [electorama/abif#28] [Announcements]
@brainbuz has made progress, but @robla hasn't.  At least until this evening. (@robla now switches out of third peson)

@brainbuz has a perl parser submitted as pull request #25 , as well as some early draft documents in pull request #24 .  I just responded on the latter request;

> I'd like to pull this after I put in the license at the top level.  I'm planning to go with the the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0.html) for software, and the [Creative Commons BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/) for the specification.  Assuming you agree with those terms (please do so explicity), I'll merge your pull request, and add you to the "ABIF Contributors" file (which hasn't been created yet).  The software and the specification will be provided by the "ABIF Contributors".

@brainbuz already responded, so now to create the LICENSE file....

**Comment by brainbuz** [comment]
Please either make the perl parser a topic branch or just leave it as a 
pending pull until its more developed.

On 1/24/22 12:52 AM, Rob Lanphier wrote:
>
> @brainbuz <https://github.com/brainbuz> has made progress, but @robla 
> <https://github.com/robla> hasn't. At least until this evening. 
> ***@***.*** <https://github.com/robla> now switches out of third peson)
>
> @brainbuz <https://github.com/brainbuz> has a perl parser submitted as 
> pull request #25 <https://github.com/electorama/abif/pull/25> , as 
> well as some early draft documents in pull request #24 
> <https://github.com/electorama/abif/pull/24> . I just responded on the 
> latter request;
>
>     I'd like to pull this after I put in the license at the top level.
>     I'm planning to go with the the Apache License 2.0
>     <https://www.apache.org/licenses/LICENSE-2.0.html> for software,
>     and the Creative Commons BY-SA 4.0
>     <https://creativecommons.org/licenses/by-sa/4.0/> for the
>     specification. Assuming you agree with those terms (please do so
>     explicity), I'll merge your pull request, and add you to the "ABIF
>     Contributors" file (which hasn't been created yet). The software
>     and the specification will be provided by the "ABIF Contributors".
>
> @brainbuz <https://github.com/brainbuz> already responded, so now to 
> create the LICENSE file....
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/electorama/abif/discussions/28>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AADOTPB7MHK37U32RS6RVU3UXTSJHANCNFSM5MUN3G4A>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>
-- 

Minds gets scrambled like eggs
Get bruised and erased
When you live in a brainstorm

— Alice Cooper: Hard Hearted Alice

**Comment by robla** [comment]
I'm not planning on merging the Perl parser just yet.  In fact, my plan is to have a full Python parser (or perhaps my own kludgy Perl parser) in a separate repository (quite possibly GPLv3 or other slightly more restrictive license).  The goal would be to get a lot of different implementations going, so that we don't leave anything out of the specification.  The Python code in this repo is the minimum amount necessary to use with the Lark parser (and [the EBNF that Lark reads](https://github.com/electorama/abif/blob/main/abif-v0.01.ebnf)).  So I'm not sure it makes sense for me to merge your Perl-based parser into this, but rather link to it on a list of implementations (or something).
